flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSource.java

```diff
@@ -98,20 +98,24 @@
     private final KafkaRecordDeserializationSchema<OUT> deserializationSchema;
     // The configurations.
     private final Properties props;
+    // Client rackId callback
+    private final Supplier<String> rackIdSupplier;
```

Comment:
This should be a "SerializableSupplier". A supplier may contain a non-serializable client for example

Reply:
You're right about the Serializable bit, good catch.

Is there a straightforward way to do that while still allowing for Lambdas? Maybe provide one method for the basic Supplier and another for ClosableSerializableSupplier for when you need that degree of safety.